tools/ci/check_grep2.py

```diff
@@ -0,0 +1,337 @@
+from pathlib import Path
+import re
+import sys
+import time
+from collections import OrderedDict
+
+RED = "\033[0;31m"
+GREEN = "\033[0;32m"
+BLUE = "\033[0;34m"
+NC = "\033[0m"  # No Color
+
+MAP_COMMENT = "//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE"
+IGNORE_515_PROC_MARKER = "__byond_version_compat.dm"
+
+
+def check_tgm(filename, lines):
+    for line in lines:
+        if re.match(r"^\".+\" = \(.+\)", line):
+            return "ERROR: Non-TGM formatted map detected. Please convert it using Map Merger!"
+
+
+def check_comments(filename, lines):
+    for line in lines:
+        if (
+            re.match(r"//", line)
```

Comment:
Since this is going to parse a Lot of code and these same expressions are going to be hit a bunch, I'm curious if there might be some time benefit to pre-compiling the patterns with `re.compile(pat)` at the start of the program. From what I've read, the performance impact might be negligible but could also offer a nonzero speedup.

Reply:
I'm compiling these now. The performance difference is probably negligible but this is cleaner regardless.